pkg/corpusaccess/corpusaccessor.py

```diff
@@ -0,0 +1,42 @@
+import yaml
+
+
+class CorpusAccessor:
+	# private construction for singleton
+	class __CorpusAccessor:
+		def __init__(self, ctx):
+			self.ctx = ctx
+			self.documents = {}
+
+			with open(self.ctx.corpus_path, "r") as corpus_handle:
+				corpus_stream = yaml.load_all(corpus_handle, Loader=yaml.Loader)
+				for doc in corpus_stream:
+					self.documents[doc.id] = doc
+	
+	corpora = {}
+	def __init__(self, ctx):
+		# if the corpus is not yet in accessor, then load the corpus and put it in corpora
+		if not ctx.corpus_path in self.corpora:
+			CorpusAccessor.corpora[ctx.corpus_path] = CorpusAccessor.__CorpusAccessor(ctx)
+			# is using the corpus_path as a dict key a bad idea?
```

Comment:
I don't quite understand why we need to wrap the accessor in a dict, can you explain that to me?

Reply:
I was trying to think ahead to when theres multiple corpora, but maybe I'm misguided. My thought was that if there was a query on the uOttawa catalogue, then on Reuters, then another one on uOttawa, that it would load in each library only once, but still be able to switch between them. And that it could differentiate them based on the corpus path of that query, since that would be unique. Does that make sense? I can change it, especially since theres only the uOttawa corpus considered for the first submission anyway.

Reply:
Ah, that makes sense! I completely hadn’t thought of that!